blocks/browse/da-browse/da-search.js

```diff
@@ -0,0 +1,162 @@
+            this._total += 1;
+            const file = await this.searchFile(item, term);
+            if (file.match) {
+              this._matches += 1;
+              this._items = [...this._items, file];
+              this.updateList();
+            }
+          }
+          if (!item.ext) return item.path;
+          return null;
+        });
+
+        const newFolders = await Promise.all(searchFolderItems);
+        return newFolders.filter(Boolean);
+      });
+      const childFolders = await Promise.all(searchFolders);
+      folders = childFolders.flat();
+    }
+  }
+
+  async search(startPath, term) {
+    this._action = 'Found';
+    performance.mark('start-search');
+    await this.getMatches(startPath, term);
+    performance.mark('end-search');
+    performance.measure('search', 'start-search', 'end-search');
+    const searchTime = performance.getEntriesByName('search')[0].duration;
+    this._time = String(searchTime / 1000).substring(0, 4);
+  }
+
+  async handleSearch(e) {
+    e.preventDefault();
+    this.setDefault();
+    const [term] = e.target.elements;
+    if (!term.value) return;
+    this._term = term.value;
+    this.search(this.path, term.value);
+  }
+
+  async handleReplace(e) {
+    e.preventDefault();
+    const [replace] = e.target.elements;
+    if (!replace.value) return;
+
+    this._time = null;
+    this._action = 'Replaced';
+    this._total = this._matches;
+    this._matches = 0;
+    performance.mark('start-replace');
+
+    const replaced = this._items.map((match) => new Promise((resolve) => {
+      (async () => {
+        const resp = await daFetch(`${DA_ORIGIN}/source${match.path}`);
+        if (!resp.ok) return;
+        const text = await resp.text();
+        const replacedText = text.replaceAll(this._term, replace.value);
+        const blob = new Blob([replacedText], { type: 'text/html' });
+        const formData = new FormData();
+        formData.append('data', blob);
+        const opts = { method: 'PUT', body: formData };
+        const daResp = await daFetch(`${DA_ORIGIN}/source${match.path}`, opts);
+        if (!daResp.ok) return;
+        this._matches += 1;
+        resolve();
+      })();
+    }));
+    await Promise.all(replaced);
+    performance.mark('end-replace');
+    performance.measure('replace', 'start-replace', 'end-replace');
+    const replaceTime = performance.getEntriesByName('replace')[0].duration;
+    this._time = String(replaceTime / 1000).substring(0, 4);
+  }
+
+  async toggleReplace() {
+    this.showReplace = !this.showReplace;
+  }
+
+  render() {
+    return html`
+      <form @submit=${this.handleSearch}>
+        <input type="text" placeholder="Enter search" name="term"/>
+        <input type="submit" value="Search" />
+      </form>
+      <p>${this.matches && this._total ? html`${this._action} ${this._matches} of ${this._total} ${this._time ? html`in ${this._time} seconds.` : nothing}` : nothing}</p>
```

Comment:
might be cleaner to assign this to a const instead of inlining

Reply:
Good idea.